cmd/utils/flags.go

```diff
@@ -663,7 +663,7 @@ func MakeNode(ctx *cli.Context, name, gitCommit string) *node.Node {
 		Version:           vsn,
 		UserIdent:         makeNodeUserIdent(ctx),
 		NoDiscovery:       ctx.GlobalBool(NoDiscoverFlag.Name) || ctx.GlobalBool(LightModeFlag.Name),
-		DiscoveryV5:       ctx.GlobalBool(DiscoveryV5Flag.Name) || ctx.GlobalBool(LightModeFlag.Name) || ctx.GlobalInt(LightServFlag.Name) > 0,
+		DiscoveryV5:       ctx.GlobalBool(DiscoveryV5Flag.Name) || (ctx.GlobalBool(LightModeFlag.Name) || ctx.GlobalInt(LightServFlag.Name) > 0) && !ctx.GlobalBool(NoDiscoverFlag.Name),
```

Comment:
I find this hard to parse for a human brain. Is it possible to reconstruct this statement so it's easier to brain-validate e.g. what effect the `NoDiscoverFlag` has on the outcome?